rest/server_context.go

```diff
@@ -628,11 +618,60 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 		if resyncRequired {
 			collectionsRequiringResync = append(collectionsRequiringResync, base.ScopeAndCollectionName{Scope: base.DefaultScope, Collection: base.DefaultCollection})
 		}
-		if err := initDataStore(bucket.DefaultDataStore(), db.IndexesAll); err != nil {
+		if err := initDataStore(bucket.DefaultDataStore(), db.IndexesAll, base.DefaultScopeAndCollectionName()); err != nil {
 			return nil, err
 		}
 	}
 
+	startOffline := base.BoolDefault(config.StartOffline, false)
+	var dbInitDoneChan chan error
+	// Initialize any required indexes
+	if len(collectionsRequiringIndexes) > 0 {
+		gsiSupported := bucket.IsSupported(sgbucket.BucketStoreFeatureN1ql)
+		if !gsiSupported {
+			return nil, errors.New("Sync Gateway was unable to connect to a query node on the provided Couchbase Server cluster.  Ensure a query node is accessible, or set 'use_views':true in Sync Gateway's database config.")
+		}
+
+		// If database has been requested to start offline, or there's an active async initialization, use async initialization
```

Comment:
Add something here about the nil case being true when persistent config is not used?